### ISSUE TITLE
Problem with extend-protocol and arrays

### ISSUE THREAD

**Issue by jakepearson** [clj-kondo/clj-kondo#2129]
**version**

`v2023.05.26`

**platform**

native from the command line (and calva)

**editor**

vscode with calva

**problem**

If I use `extend-protocol` with a type that looks like `(class (float-array 0))`. I get the following error "Function arguments should be wrapped in vector.clj-kondo(syntax)". I do not think this is expected, but please correct me if I'm wrong.

**repro**

The following code in a namespace will generate the above error.

```clojure
(defprotocol do-a-thing
 (print-thing [thing]))

(extend-protocol do-a-thing
 (class (float-array 0))
  (print-thing [_]
   (println "it's a float array")))
```

![image](https://github.com/clj-kondo/clj-kondo/assets/325306/ca87ec63-775e-41ad-91d7-74616c0c2413)

**config**

This happens with and without a config file defined.

```text
❯ clj-kondo --lint src/myfile.clj
test/voltron/maths/query_test.clj:19:10: error: Function arguments should be wrapped in vector.
linting took 44ms, errors: 1, warnings: 0
```

**expected behavior**

I believe this should not issue an error or warning.

Happy to help on a fix.

**Comment by borkdude** [comment]
The behavior of using dynamic class in extend-protocol is not officially supported and fragile, see https://clojure.atlassian.net/browse/CLJ-1381.

As such, I think directly using `extend` is a better fit when you want to use a class dynamically:

``` clojure
(defprotocol Body
  (to-body [x]))

(extend (class (byte-array 0))
  Body {:to-body (fn [this] (RequestBody/fromBytes this))})
```

More info in this Clojurians Slack thread:

https://clojurians.slack.com/archives/CHY97NXE2/p1677266201633939

**Comment by jakepearson** [comment]
Thanks, I'll take a look!
